library/src/main/java/com/novoda/downloadmanager/MigrationExtractor.java

```diff
@@ -2,14 +2,12 @@
 
 import android.database.Cursor;
 
-import com.novoda.notils.logger.simple.Log;
-
 import java.util.ArrayList;
 import java.util.List;
 
 class MigrationExtractor {
 
-    private static final String BATCHES_QUERY = "SELECT batches._id, batches.batch_title FROM batches INNER JOIN DownloadsByBatch ON " +
+    private static final String BATCHES_QUERY = "SELECT DISTINCT batches._id, batches.batch_title FROM batches INNER JOIN DownloadsByBatch ON " +
```

Comment:
Here's the fix for bug one - it was effectively returning two rows for each batch, hence double the required number of migrations were occurring